src/main/java/edu/hm/hafner/analysis/IssuesFilter.java

```diff
@@ -0,0 +1,133 @@
+package edu.hm.hafner.analysis;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class IssuesFilter {
+
+    private List<FilterRegex> fileNameFilters = new ArrayList<FilterRegex>();
+    private List<FilterRegex> packageNameFilters = new ArrayList<FilterRegex>();
+    private List<FilterRegex> moduleNameFilters = new ArrayList<FilterRegex>();
+    private List<FilterRegex> categoryFilters = new ArrayList<FilterRegex>();
+    private List<FilterRegex> typeFilters = new ArrayList<FilterRegex>();
+
+    /**
+     * This represents the main functionality of the instance. Will evaluate all rules that where added by {@link
+     * #addFilenameFilter(FilterRegex)} {@link #addPackageNameFilter(FilterRegex)} {@link
+     * #addModuleNameFilter(FilterRegex)} {@link #addCategoryFilter(FilterRegex)} {@link #addTypeFilter(FilterRegex)}
+     *
+     * @param issues
+     *         list of issues that should be evaluated
+     *
+     * @return filtered list of {@link Issues}
+     */
+    Issues apply(final Issues issues) {
+        HashSet<UUID> idsToRemove = new HashSet<>();
+        for (Issue issue : issues) {
+            // FileName
+            for (FilterRegex filter : fileNameFilters) {
+                boolean matches = issue.getFileName().matches(filter.getExpression());
+                if ((filter instanceof ExcludeFilterRegex && matches) || (filter instanceof IncludeFilterRegex && !matches)) {
```

Comment:
`instanceof` Checks sollten in OO Programmen nicht erforderlich sein -> falsche Abstraktion? Wäre da ein ENUM sinnvoller?